047-TrafficControlWithDapr/Student/Challenge-07.md

```diff
@@ -0,0 +1,61 @@
+# Challenge 7 - Add secrets management
+
+[< Previous Challenge](./Challenge-06.md) - **[Home](../README.md)** - [Next Challenge >](./Challenge-08.md)
+
+## Introduction
+
+In this challenge, you're going to add the Dapr **secrets management** building block.
+
+## Description
+
+Almost all non-trivial applications need to _securely_ store secret data like API keys, database passwords, and more. As a best practice, secrets should never be checked into the version control system. But at the same time, they need to be accessible to code running in production. This is generally a challenging requirement, but critical to get right.
+
+Dapr provides a solution to this problem: The Dapr secrets building block. It includes an API and a secrets store.
+
+Here's how it works:
+
+- Dapr exposes a **secret store** - a place to securely store secret data.
+- Dapr provides an API that enables applications to retrieve secrets from the store.
+
+Popular secret stores include `Kubernetes`, `Hashicorp Vault`, and `Azure KeyVault`.
+
+The following diagram depicts an application requesting the secret called "mysecret" from a secret store called "vault" from a configured cloud secret store:
+
+<img src="../images/Challenge-07/secrets_cloud_stores.png" style="zoom:67%;" />
+
+Note the blue-colored Dapr secrets building block that sits between the application and secret stores.
+
+*For this challenge you'll use a file-based local secret store component. Local stores are meant for development or testing purposes. Never use them in production!*
+
+*Alternatively, you can implement Azure KeyVault as your secret store in this challenge.*
+
+Another way of using secrets, is to reference them from Dapr configuration files. You will use both approaches in this challenge.
+
+To learn more about the secrets building block, read the [introduction to this building block](https://docs.dapr.io/developing-applications/building-blocks/secrets/) in the Dapr documentation. Also, checkout the [secrets chapter](https://docs.microsoft.com/dotnet/architecture/dapr-for-net-developers/secrets) in the [Dapr for .NET Developers](https://docs.microsoft.com/dotnet/architecture/dapr-for-net-developers/) guidance eBook.
+
+## Success Criteria
+
+To complete this challenge, you must reach the following goals:
+
+- The credentials used by the SMTP output binding to connect to the SMTP server are retrieved using the Dapr secrets management building block.
+- The `FineCollectionService` retrieves the license key for the `FineCalculator` component it uses from the Dapr secrets management building block.
+
+This challenge targets the operation labeled as **number 6** in the end-state setup:
+
+**Local**
+
+<img src="../images/Challenge-07/secrets-management-operation.png" style="zoom: 67%;" />
```

Comment:
Could you remove "Assignment #7" from the image? (I would not replace it with "Challenge 7" in the image in case these challenges get renumbered in the future or else you'll have to do images update in the future.)